### ISSUE TITLE
consistent-type-imports fixStyle is incompatible with @typescript-eslint/no-import-type-side-effects

### ISSUE THREAD

**Issue by bigpopakap** [EvgenyOrekhov/eslint-config-hardcore#688]
With `typescript-eslint` monorepo packages starting with version `5.50.0`, I'm getting the following on lines that have only a single type import
```
// Error: TypeScript will only remove the inline type specifiers which will leave behind a side effect import at runtime. Convert this to a top-level type qualifier to properly remove the entire import
import { type Something } from './somewhere';
```

Could we update `fixStyle` to `'separate-type-imports'` to solve this issue here? A quick fix in this package would be great.

There is an issue that will fix it directly from `@typescript-eslint/consistent-type-imports`, but that may take a while: https://github.com/typescript-eslint/typescript-eslint/issues/6379

**Comment by EvgenyOrekhov** [comment]
Thanks for reporting it! Fixed and released in [v30.0.0](https://github.com/EvgenyOrekhov/eslint-config-hardcore/releases/tag/v30.0.0)

**Comment by bigpopakap** [comment]
Amazing, thank you!
